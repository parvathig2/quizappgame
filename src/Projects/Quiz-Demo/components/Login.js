import React from 'react'

const Login = ({
    selectedLang,setSelectedLang,
    userName,setUserName,
    handleQuizLogic,displayError,
    selectedGender,setSelectedGender,error}) => {

      const handleNameChange = (e)=>{
          setUserName(e.target.value)
      }

      const handleLangChange = (e)=>{
          setSelectedLang(e.target.value)
         
      }
      const handleCatChange = (e)=>{
          
          setSelectedGender(e.target.value)
      }

      const handleClick = ()=>{
         
         (userName && selectedLang) ?  handleQuizLogic(): displayError();

      }
      
    
    return <div className='wrapper'>
        <div className='form'>
            {/* {error && alert('name cannot be blank')} */}
            <input type="text" autoComplete='off'  value={userName} onChange={handleNameChange} placeholder='Enter your name here' ></input>
            <select value={selectedGender} name="category" id="category" onChange={handleCatChange}>
                <option value="male">male</option>
                <option value="female">female</option>
                <option value="others">others</option>
            </select>

            <select value={selectedLang} name="languages" id="languages" onChange={handleLangChange}>
                <option value="english">english</option>
                <option value="kannada">kannada</option>
                <option value="hindi">hindi</option>
            </select>
            <button className='btn' onClick={handleClick}>start quiz</button>
           
               
           
            
        </div>
        
    </div>
}

export default Login
