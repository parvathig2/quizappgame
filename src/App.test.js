import React from "react";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import Login from "./Projects/Quiz-Demo/components/Login";
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'


cleanup()
test('should rednder the login',()=>{
    render(<Login></Login>);
    const value = screen.getByText("Login");
    expect(value).toBeInTheDocument()
    

    
 
})

describe("",()=>{
    test("test on rendering  input HomePage by sending dummyprop ", () => {
        render(
         
            <Login
              userName={"varun"}
              selectedLang={"english"}
            ></Login>
        
        );
    
        expect(screen.getByDisplayValue("varun")).toBeInTheDocument();
        expect(screen.getByDisplayValue("english")).toBeInTheDocument();

        const tree = renderer
      .create(
        
          <Login
           userName={"varun"}
           selectedLang={"english"}
          ></Login>
        
      )
      .toJSON();
    console.log("::", tree);
    expect(tree).toMatchSnapshot();
})
});


