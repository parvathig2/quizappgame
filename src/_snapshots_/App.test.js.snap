// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` test on rendering  input HomePage by sending dummyprop  1`] = `
<div
  className="wrapper"
>
  <h4
    data-testid="login"
  >
    Login
  </h4>
  <div
    className="form"
  >
    <input
      autoComplete="off"
      name="name"
      onChange={[Function]}
      type="text"
      value="varun"
    />
    <select
      id="category"
      name="category"
      onChange={[Function]}
    >
      <option
        value="male"
      >
        male
      </option>
      <option
        value="female"
      >
        female
      </option>
      <option
        value="others"
      >
        others
      </option>
    </select>
    <select
      id="languages"
      name="languages"
      onChange={[Function]}
      value="english"
    >
      <option
        value="english"
      >
        english
      </option>
      <option
        value="kannada"
      >
        kannada
      </option>
      <option
        value="hindi"
      >
        hindi
      </option>
    </select>
    <button
      className="btn"
      onClick={[Function]}
    >
      start quiz
    </button>
  </div>
</div>
`;
